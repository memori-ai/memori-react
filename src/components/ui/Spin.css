.memori-spin {
    position: relative;
}

.memori-spin--spinner {
    position: absolute;
    z-index: 100;
    top: 0;
    left: 0;
    display: none;
    width: 100%;
    height: 100%;
    align-items: center;
    justify-content: center;
    background: rgba(255, 255, 255, 0.8);
}

.memori-spin--spinning .memori-spin--spinner {
    display: flex;
}

.memori-spin--spinner svg {
    width: 30%;
    max-width: 5rem;
    height: 30%;
    max-height: 5rem;
}

.memori-spin--primary svg {
    fill: var(--memori-primary);
}
