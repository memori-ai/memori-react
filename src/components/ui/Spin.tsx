import React, { FC } from 'react';
import cx from 'classnames';
import Loading from '../icons/Loading';

export interface Props {
  spinning?: boolean;
  className?: string;
  primary?: boolean;
  children?: JSX.Element | React.ReactNode;
}

const Spin: FC<Props> = ({
  spinning = false,
  primary = false,
  className,
  children,
}: Props) => (
  <div
    className={cx('memori-spin', className, {
      'memori-spin--spinning': spinning,
      'memori-spin--primary': primary,
    })}
  >
    {children}
    <div className="memori-spin--spinner">
      <Loading loading />
    </div>
  </div>
);

export default Spin;
