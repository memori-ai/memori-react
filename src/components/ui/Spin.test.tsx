import React from 'react';
import { render } from '@testing-library/react';
import Spin from './Spin';

const content = (
  <>
    <p>Lorem ipsum dolor sit amet, consectetur adipiscing elit.</p>
    <h2>Suspendisse a sodales nulla, sed semper nisi.</h2>
    <p>Proin tincidunt enim in felis aliquet, a ultricies purus bibendum.</p>
    <ul>
      <li>Quisque in ultrices lectus.</li>
      <li>Quisque in ultrices lectus.</li>
      <li>Quisque in ultrices lectus.</li>
    </ul>
    <p>Nulla at urna diam.</p>
  </>
);

it('renders Spin unchanged', () => {
  const { container } = render(<Spin>{content}</Spin>);
  expect(container).toMatchSnapshot();
});

it('renders Spin spinning unchanged', () => {
  const { container } = render(<Spin spinning>{content}</Spin>);
  expect(container).toMatchSnapshot();
});

it('renders Spin with custom className unchanged', () => {
  const { container } = render(<Spin className="lorem-ipsum">{content}</Spin>);
  expect(container).toMatchSnapshot();
});

it('renders Spin spinning primary unchanged', () => {
  const { container } = render(
    <Spin spinning primary>
      {content}
    </Spin>
  );
  expect(container).toMatchSnapshot();
});
