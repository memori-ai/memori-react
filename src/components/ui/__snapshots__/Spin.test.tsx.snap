// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Spin spinning primary unchanged 1`] = `
<div>
  <div
    class="memori-spin memori-spin--spinning memori-spin--primary"
  >
    <p>
      Lorem ipsum dolor sit amet, consectetur adipiscing elit.
    </p>
    <h2>
      Suspendisse a sodales nulla, sed semper nisi.
    </h2>
    <p>
      Proin tincidunt enim in felis aliquet, a ultricies purus bibendum.
    </p>
    <ul>
      <li>
        Quisque in ultrices lectus.
      </li>
      <li>
        Quisque in ultrices lectus.
      </li>
      <li>
        Quisque in ultrices lectus.
      </li>
    </ul>
    <p>
      Nulla at urna diam.
    </p>
    <div
      class="memori-spin--spinner"
    >
      <svg
        aria-hidden="true"
        class="memori-loading-icon"
        focusable="false"
        role="img"
        viewBox="0 0 1024 1024"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M988 548c-19.9 0-36-16.1-36-36 0-59.4-11.6-117-34.6-171.3a440.45 440.45 0 00-94.3-139.9 437.71 437.71 0 00-139.9-94.3C629 83.6 571.4 72 512 72c-19.9 0-36-16.1-36-36s16.1-36 36-36c69.1 0 136.2 13.5 199.3 40.3C772.3 66 827 103 874 150c47 47 83.9 101.8 109.7 162.7 26.7 63.1 40.2 130.2 40.2 199.3.1 19.9-16 36-35.9 36z"
        />
      </svg>
    </div>
  </div>
</div>
`;

exports[`renders Spin spinning unchanged 1`] = `
<div>
  <div
    class="memori-spin memori-spin--spinning"
  >
    <p>
      Lorem ipsum dolor sit amet, consectetur adipiscing elit.
    </p>
    <h2>
      Suspendisse a sodales nulla, sed semper nisi.
    </h2>
    <p>
      Proin tincidunt enim in felis aliquet, a ultricies purus bibendum.
    </p>
    <ul>
      <li>
        Quisque in ultrices lectus.
      </li>
      <li>
        Quisque in ultrices lectus.
      </li>
      <li>
        Quisque in ultrices lectus.
      </li>
    </ul>
    <p>
      Nulla at urna diam.
    </p>
    <div
      class="memori-spin--spinner"
    >
      <svg
        aria-hidden="true"
        class="memori-loading-icon"
        focusable="false"
        role="img"
        viewBox="0 0 1024 1024"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M988 548c-19.9 0-36-16.1-36-36 0-59.4-11.6-117-34.6-171.3a440.45 440.45 0 00-94.3-139.9 437.71 437.71 0 00-139.9-94.3C629 83.6 571.4 72 512 72c-19.9 0-36-16.1-36-36s16.1-36 36-36c69.1 0 136.2 13.5 199.3 40.3C772.3 66 827 103 874 150c47 47 83.9 101.8 109.7 162.7 26.7 63.1 40.2 130.2 40.2 199.3.1 19.9-16 36-35.9 36z"
        />
      </svg>
    </div>
  </div>
</div>
`;

exports[`renders Spin unchanged 1`] = `
<div>
  <div
    class="memori-spin"
  >
    <p>
      Lorem ipsum dolor sit amet, consectetur adipiscing elit.
    </p>
    <h2>
      Suspendisse a sodales nulla, sed semper nisi.
    </h2>
    <p>
      Proin tincidunt enim in felis aliquet, a ultricies purus bibendum.
    </p>
    <ul>
      <li>
        Quisque in ultrices lectus.
      </li>
      <li>
        Quisque in ultrices lectus.
      </li>
      <li>
        Quisque in ultrices lectus.
      </li>
    </ul>
    <p>
      Nulla at urna diam.
    </p>
    <div
      class="memori-spin--spinner"
    >
      <svg
        aria-hidden="true"
        class="memori-loading-icon"
        focusable="false"
        role="img"
        viewBox="0 0 1024 1024"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M988 548c-19.9 0-36-16.1-36-36 0-59.4-11.6-117-34.6-171.3a440.45 440.45 0 00-94.3-139.9 437.71 437.71 0 00-139.9-94.3C629 83.6 571.4 72 512 72c-19.9 0-36-16.1-36-36s16.1-36 36-36c69.1 0 136.2 13.5 199.3 40.3C772.3 66 827 103 874 150c47 47 83.9 101.8 109.7 162.7 26.7 63.1 40.2 130.2 40.2 199.3.1 19.9-16 36-35.9 36z"
        />
      </svg>
    </div>
  </div>
</div>
`;

exports[`renders Spin with custom className unchanged 1`] = `
<div>
  <div
    class="memori-spin lorem-ipsum"
  >
    <p>
      Lorem ipsum dolor sit amet, consectetur adipiscing elit.
    </p>
    <h2>
      Suspendisse a sodales nulla, sed semper nisi.
    </h2>
    <p>
      Proin tincidunt enim in felis aliquet, a ultricies purus bibendum.
    </p>
    <ul>
      <li>
        Quisque in ultrices lectus.
      </li>
      <li>
        Quisque in ultrices lectus.
      </li>
      <li>
        Quisque in ultrices lectus.
      </li>
    </ul>
    <p>
      Nulla at urna diam.
    </p>
    <div
      class="memori-spin--spinner"
    >
      <svg
        aria-hidden="true"
        class="memori-loading-icon"
        focusable="false"
        role="img"
        viewBox="0 0 1024 1024"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M988 548c-19.9 0-36-16.1-36-36 0-59.4-11.6-117-34.6-171.3a440.45 440.45 0 00-94.3-139.9 437.71 437.71 0 00-139.9-94.3C629 83.6 571.4 72 512 72c-19.9 0-36-16.1-36-36s16.1-36 36-36c69.1 0 136.2 13.5 199.3 40.3C772.3 66 827 103 874 150c47 47 83.9 101.8 109.7 162.7 26.7 63.1 40.2 130.2 40.2 199.3.1 19.9-16 36-35.9 36z"
        />
      </svg>
    </div>
  </div>
</div>
`;
